mempool/mempool.go

```diff
@@ -0,0 +1,231 @@
+package mempool
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/encoder"
+)
+
+type ValidatorFunc func(*BroadcastedTransaction) error
+
+type BroadcastedTransaction struct {
+	Transaction   core.Transaction
+	DeclaredClass core.Class
+}
+
+const (
+	poolLengthKey = "poolLength"
+	headKey       = "headKey"
+	tailKey       = "tailKey"
+)
+
+// Pool stores the transactions in a linked list for its inherent FCFS behaviour
+type storageElem struct {
+	Txn      BroadcastedTransaction
+	NextHash *felt.Felt
+}
+
+type Pool struct {
+	db        db.DB
+	validator ValidatorFunc
+	txPushed  chan struct{}
+}
+
+func New(poolDB db.DB) *Pool {
+	return &Pool{
+		db:        poolDB,
+		validator: func(_ *BroadcastedTransaction) error { return nil },
+		txPushed:  make(chan struct{}, 1),
+	}
+}
```

Comment:
This is an example of what we talked in the daily regarding escape analysis. When returning a reference to `Pool` you are allocating it into the the heap, if you change the result to `Pool` it will be stored in the stack.

You can see the escape analysis made by the compiler using this command
```
go build -gcflags="-m" ./mempool/mempool.go"
```

Which will show a line like this in the output:
```
...
... some compiler analysis
...

mempool/mempool.go:39:9: &Pool{...} escapes to heap

...
... other variables that escape or not
...
```